units/patch_intent.go

```diff
@@ -483,6 +493,46 @@ func (j *patchIntentProcessor) getPreviousPatchDefinition(project *model.Project
 	return res, nil
 }
 
+func getPreviousFailedTasksAndDisplayTasks(tasksInProjectVariant []string, displayTasksInProjectVariant []string, vt patch.VariantTasks, version string) ([]string, []patch.DisplayTask, error) {
+	failedTasks, err := task.FindAll(db.Query(task.FailedExecutionTasksByVersionAndBV(version, vt.Variant)))
```

Comment:
Do we need to query for tasks/display tasks separately? Can we query all of them together, and then when we loop through them, either add them to failedTaskNames or add them to a list of failedDisplayTasks?

Reply:
Also, should these only find the tasks that we specifically defined in vt?

Reply:
I'm now using utility.StringSliceIntersection on both string arrays to ensure this although this could also be done in the DB operation